Userland/Libraries/LibWeb/Page/Frame.h

```diff
@@ -101,6 +103,10 @@ class Frame : public TreeNode<Frame> {
 
     void did_edit(Badge<EditEventHandler>);
 
+    size_t get_recursion_level() { return m_recursion_level; }
```

Comment:
In Firefox its not really clear to me how many iframes are allowed to render. In my simple test file which just includes itself I get a maximum of 3, opening https://cbracco.github.io/html5-test-page/ I get 4 as the deepest level. It doesn't matter that much however, as long as it's a finite number. What would you prefer?

`frame_nesting_level`sounds good, way more descriptive of what it's actually used for, I'll change it to that!

Reply:
So including yourself is one thing, but what if you have a deeply nested set of frames that are actually different files?

Reply:
It seems you can nest as many iframes as you want as long as they all have different `src`. The recursion protection only hits if you try to include an iframe you have already included twice in the stack.
So we have to track every nested iframe and its src for the page and only bail out when appropriate.

I'll see what I can come up with.

Reply:
Okay, now every Frame keeps track of the nesting in a HashMap, where for each URL the nesting level is recorded. The Map gets passed from the parent to the children when a new iframe is created. 

I don't think this can be done in a more central way as multiple iframes with the same src on the same page should be no problem (and this way they aren't).